include/nlohmann/detail/conversions/to_chars.hpp

```diff
@@ -1062,7 +1063,7 @@ format. Returns an iterator pointing past-the-end of the decimal representation.
 template<typename FloatType>
 JSON_HEDLEY_NON_NULL(1, 2)
 JSON_HEDLEY_RETURNS_NON_NULL
-char* to_chars(char* first, const char* last, FloatType value)
+char* to_chars(char* first, const char* last, FloatType value, size_t precision = std::numeric_limits<FloatType>::max_digits10)
```

Comment:
Just to understand: why did you provide a default value for `precision` here?

Reply:
because this is the largest meaningful precision, I will use this as default throughtout

Reply:
I would suggest `int precisionOverride = -1` and any value less than 0 is ignored and the current behavior is used.